ci-operator/step-registry/ovn/ex-gw-e2e/ovn-ex-gw-e2e-commands.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ ovn external gateways e2e commands ************"
+
+# shellcheck source=/dev/null
+source "${SHARED_DIR}/packet-conf.sh"
+
+#OVNK_SRC_DIR="/go/src/github.com/openshift/ovn-kubernetes"
+
+#if [ -d "${OVNK_SRC_DIR}" ]; then
+#  echo "### Copying ovnk directory"
+#  scp "${SSHOPTS[@]}" -r "${OVNK_SRC_DIR}" "root@${IP}:/root/dev-scripts/ovnk/"
+#fi
+
+# TMP - Cloning manually - will be replaced by scp^^
+echo "### Cloning ovnk"
+ssh "${SSHOPTS[@]}" "root@${IP}" "cd /root/dev-scripts/ && git clone https://github.com/ovn-org/ovn-kubernetes.git && cd ovn-kubernetes && git fetch origin pull/2791/head:tmp_gw && git checkout tmp_gw"
```

Comment:
What does this do? If you need to run some code from ovn-k, it's better to run it as an explicit command / script from the release job.

Check out https://github.com/openshift/release/blob/master/ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-master.yaml#L98 for an example.

Reply:
The goal is to run the tests from a host (with docker/podman installed) that can reach the ci bare-metal cluster. I think (not sure) running it the way you linked runs it in a pod inside the cluster.
So what we do here is cloning it directly from the vm that installed the cluster and is in the same network. Later on the script we run the e2e directly on that vm with ssh.

This is only a workaround for now, when the changes from the specific PR land upstream and downstream we could copy the files with scp without having to clone from inside the vm - pretty similar to how the current MetalLB tests work.

Reply:
The way I linked also runs as a pod inside the CI cluster. Is that sufficient? That pod can then create arbitrary pods inside the cluster under test. So you should be able to do anything.

Reply:
The e2e that I'm running creates external containers and sets them up as external gateways. I want to preserve that without changing the code too much, so I need to run them on a host that I can create containers on, using the host network https://github.com/ovn-org/ovn-kubernetes/pull/2791.

Reply:
Hm. Gotcha. Ideally the upstream tests would just create HostNetwork pods on the cluster under test. Maybe we can move in that direction?

Reply:
They could, however the tests also create endpoints and listeners on the external containers `lo` device and sets route back to the src pods. Moving it to create pods instead would mean the traffic back to the src pods would be handled differently I think (by ovnk) and could end up as a pretty big change - not sure if covering the same stuff.

Reply:
Would they? They would be HostNetwork, and thus the same thing as `docker exec kind-node`, from a networking perspective...

Reply:
yeah changing the external gateway pods to host-networked is what I had suggested as well because technically speaking the exgw pods are always inside the cluster not outside as "external containers" but I think @fedepaol said that's not accurate for the testing or something? It does feel a bit weird that we hack the secondary interface of the exgw pod as an external container.

side note: We do have that bug where using host-networked pods as exgw hops results in asymmetric traffic path. Irrespective of that, we still need a way to do external->cluster type e2e tests, so this can be useful, but I still think the exgw pods are usually within the cluster so using what @squeed suggested is not a bad idea though here it will involve major refactor of the existing e2e's...

I also know that @kyrtapz is working on a similar testing setup for checking external -> cluster traffic for egressIPs.